test/infrastructure/docker/.gitignore

```diff
@@ -1,20 +1,67 @@
-# invalid samples.
-config/samples
+# Binaries for programs and plugins
+*.exe
```

Comment:
I'm not against leaving those exceptions, but It seems to me that some of them does not apply to the docker provider. e.g.
- `exe`, `so` etc are type of files that are never generated in `test/infrastructure/docker`
- `cmd/clusterctl` or `bin` or `hack/tools` (or others) are top-level folders, and they don't exists under `test/infrastructure/docker`